### PR TITLE
Create the "Services naming convention" ADR

### PR DESCRIPTION
**Pull request by jakubtobiasz**
Every opinion on this is welcome :).

### CODE REVIEW

File: adr/2024_02_24_services_naming_convention.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# Remove flush from handlers
```
Comment (NoResponseMate) [inline adr/2024_02_24_services_naming_convention.md]: The what now

**Review by Rafikooo — DISMISSED** [review]

File: adr/2024_02_24_services_naming_convention.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+### Use the FQCN as a service name
+
+This variant of naming appeared along with introducing the `autowire` feature in Symfony.
+It makes using services easier, but at the same time introduces a little bit magic to the code.
+
+* Good, because it supports the `autowire` feature
+* Bad, because it requires a lot of work from the developers already using the `dot` notation
+* Bad, because it is not recommended by Symfony
+
+### Combine the `dot` notation with the FQCN (when it makes sense)
```
Comment (NoResponseMate) [inline adr/2024_02_24_services_naming_convention.md:36]: Write a bit more about """(when it makes sense)"""

File: adr/2024_02_24_services_naming_convention.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+* Bad, because it requires more work and increases the number of service definitions
+
+## Decision Outcome
+
+Chosen option: **"Combine the `dot` notation with the FQCN (when it makes sense)"**
+
+Despite the fact that it requires more work, it is the best option to provide a consistent way of naming services and support the `autowire` feature.
+Also, thanks to this approach, we stay consistent with the Symfony best practices.
+
+Some services that are not meant to be used with the `autowire` feature can be should be named with the `dot` notation.
```
Comment (NoResponseMate) [inline adr/2024_02_24_services_naming_convention.md]:
```suggestion
Some services that are not meant to be used with the `autowire` feature should only be named with the `dot` notation.
```

File: adr/2024_02_24_services_naming_convention.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+Example:
+- form types/extensions
```
Comment (NoResponseMate) [inline adr/2024_02_24_services_naming_convention.md]:
```suggestion
Some services that are not meant to be used with the `autowire` feature can be should be named with the `dot` notation:
- form types/extensions
```
Having Example twice one after another looks weird.

**Comment by mbabker** [comment]
I'd stick with dot notation for all service IDs and define FQCN aliases for selected services.

IMO, the issue with using FQCN service IDs is that it implicitly turns every service into an autowire-able service, and not every service is meant to be autowire-able (or even publicly consumed, remember there isn't a notion of package-private services in the container).  In my case with Pagerfanta, downstream users aren't meant to directly inject view classes anywhere as a service, but instead resolve those through the view factory.  The view factory has a FQCN alias to support this workflow.  For Sylius, I doubt that `Sylius\Component\Core\OrderProcessing\OrderAdjustmentsClearer` is meant to be used on its own, but instead should be called as part of the composite order processor service (whose interface, `Sylius\Component\Order\Processor\OrderProcessorInterface`, is defined as a FQCN alias for autowiring).

File: adr/2024_02_24_services_naming_convention.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
+# Remove flush from handlers
+
+* Status: drafted
+* Date: 2024-02-24
+
+## Context and Problem Statement
+
+Services in Sylius are named in different ways, depending on the time of their creation, and other services' names.
+Also, some services named with using the `dot` notation are harder to be used with using the `autowire` feature.
```
Comment (diimpp) [inline adr/2024_02_24_services_naming_convention.md]:
```suggestion
Also, some services named with using the `dot` notation are harder to be used with `autowire` feature.
```

File: adr/2024_02_24_services_naming_convention.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+## Decision Drivers
+
+* make naming services predictable
+* provide a consistent way of naming services
+* support the `autowire` feature
```
Comment (diimpp) [inline adr/2024_02_24_services_naming_convention.md]:
```suggestion
* Make naming services predictable
* Provide a consistent way of naming services
* Support the `autowire` feature
```

**Comment by Wojdylak** [comment]
I am closing this PR. It will be implemented in #17093

**Pull request closed without merging**